BWAPI4J/src/main/java/org/openbw/bwapi4j/Player.java

```diff
@@ -706,9 +706,14 @@ public boolean canMake(UnitType type) {
   }
 
   public boolean canMake(Unit builder, UnitType type) {
-    if (!canMake(type) || !builder.isA(type.whatBuilds().getFirst())) {
-      return false;
-    }
+	  if (!canMake(type) 
```

Comment:
Since `builder` is a `Unit`, you can get rid of the three checks for "is a hatchery or is a lair or is a hive" and just write `builder instanceof ZergResourceDepot` which reduces the amount of code. Also, what is the purpose of checking `whatBuilds` for `UnitType.Zerg_Larva`? Can we not omit that check?

Reply:
The check is intended to allow the case where `type` builds from `Zerg_Larva` and the `builder` is a `ZergResourceDepot` to pass through without returning false. Removing that check would allow all cases where `builder` is a `ZergResourceDepot` to go through without returning false. 

Thought I'd keep the checks against `UnitType` since that seemed to be the style - will revise to `instanceof ZergResourceDepot`

Reply:
If I understand correctly, you're saying that `self.canMake(myHatchery, UnitType.Zerg_Hydralisk` will return true with your change? Then, I assume all other production facilities will need to be considered as well for cases such as `self.canMake(myFactory, UnitType.Terran_Goliath)`?

Reply:
Yes - the conditional is intended to filter out any invalid `builder` / `type` combos. Everything else passes through fine, but Zerg `MobileUnits` are an exception because they are paired with `Zerg_Larva` and would get filtered out when combined with a `ZergResourceDepot`